lib/sqlalchemy/orm/clsregistry.py

```diff
@@ -239,10 +239,10 @@ def _remove_item(self, ref: weakref.ref[Type[Any]]) -> None:
     def add_item(self, item: Type[Any]) -> None:
         # protect against class registration race condition against
         # asynchronous garbage collection calling _remove_item,
-        # [ticket:3208]
+        # [ticket:3208] and [ticket:10782]
         modules = {
             cls.__module__
-            for cls in [ref() for ref in self.contents]
+            for cls in [ref() for ref in list(self.contents)]
```

Comment:
**Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5031